### ISSUE TITLE
Frontmatter Support

### ISSUE THREAD

**Issue by cjkoepke** [tgreyuk/typedoc-plugin-markdown#384]
Hi there. It seems like this theme includes support for generating frontmatter, but fails to actually inject it into the generated markdown files. Wondering if I'm doing something wrong or if this is expected/incomplete behavior.

Further, what is the best way to add this if it's not already implemented?

**Comment by tgreyuk** [comment]
This is available in next version (typedoc-plugin-markdown@next)

Some docs here: https://github.com/tgreyuk/typedoc-plugin-markdown/blob/next/packages/typedoc-plugin-markdown/docs/frontmatter.md

**Comment by cjkoepke** [comment]
Awesome, this is working great. Only thing I noticed is that when using a frontmatter tag with an underscore, it does not close the tags. i.e.

```ts
/**
 * @title Example
 */
```
 
will create
 
```yaml
---
title: Example
---
```

But doing something like:

```ts
/**
 * @title Example
 * @nav_order 1
 */
```

will produce invalid markup:

```yaml
---
title: Example
@nav_order 1
```

**Comment by tgreyuk** [comment]
Thanks for raising this and apologies for the delayed response.

Interestingly underscores in a tagNames in tsdoc is not valid and tags with underscores are not getting recognised properly.

Therefore the way to address this I think is to write the tag as camelCase and expose an additional option (`--frontmatterNamingConvention`) that formats to the required naming convention. I have added a few notes here: https://github.com/tgreyuk/typedoc-plugin-markdown/blob/next/packages/typedoc-plugin-markdown/docs/frontmatter.md#front-matter-variable-naming-convention.

The option is available in "typedoc-plugin-markdown@4.0.0-next.5".

**Comment by tgreyuk** [comment]
Frontmatter support now moved to seperate plugin: - [typedoc-plugin-frontmatter ](https://github.com/tgreyuk/typedoc-plugin-frontmatter). 

Any issues found please raise there. Closing this issue.
